source/fundamentals/connection.txt

```diff
@@ -4,16 +4,185 @@ Connection Guide
+image explains each part of a sample connection URI:
+
+.. figure:: /includes/figures/connection_string_parts.png
+   :alt: Each part of the connection string
+
+In this example, we use ``mongodb+srv`` for the protocol, which specifies the
+:manual:`DNS Seedlist Connection Format
+</reference/connection-string/#std-label-connections-dns-seedlist>`. This indicates
+that the URI's hostname corresponds to the DNS SRV record of your
+MongoDB instance or deployment. If your instance or deployment does not have a
+DNS SRV record, use ``mongodb`` to specify the :manual:`Standard Connection
+Format </reference/connection-string/#standard-connection-string-format>`.
+
+.. note::
+
+   If your deployment is on MongoDB Atlas, see the
+   :atlas:`Atlas driver connection guide </driver-connection>`
+   and select Go from the language dropdown to retrieve your connection string.
+
+The next part of the connection string contains your username and, if
+you are using password-based authentication, your password. Replace the value of
+``user`` with your username and ``pass`` with your password. If you are using an
+authentication mechanism that does not require a username and password, omit
+this section of the connection URI.
+
+The next part of the connection string specifies the hostname or IP address and
+port of your MongoDB instance. In the example above, we use ``sample-hostname``
+as the hostname and ``27017``, the default port, as the port. Replace these values to point to
+your MongoDB instance.
+
+The last part of the connection string specifies connection and authentication
+options. In the example, we set two connection options:
+``poolSize=20`` and ``writeConcern=majority``. For more information on connection
+options, read the :ref:`connection-options` section of this guide.
+
+The code below shows how you can use the sample connection URI in a client to
+connect to MongoDB.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/srv.go
+   :language: go
+
+Connect to ``localhost``
+++++++++++++++++++++++++
+
+.. include:: /includes/localhost-connection.rst
+
+------------------------
+Connect to a Replica Set
+------------------------
+
+A MongoDB replica set deployment is a group of connected instances that
+store the same set of data. This configuration provides data
+redundancy and high data availability.
+
+To connect to a replica set deployment, specify the hostname and port numbers
+of each instance, separated by commas, and the replica set name as the value
+of the ``replicaSet`` parameter in the connection string. In the following
+example, the replica set name is ``myRS``.
+
+.. code-block:: none
+
+   mongodb://host1:27017,host2:27017,host3:27017/?replicaSet=myRS
+
+When connecting to a replica set, the driver takes the following actions by default:
+
+- Discovers all replica set members when given the address of any one member.
+- Dispatches operations to the appropriate member, such as a directive
```

Comment:
Question: What do you mean by "a directive"?

Reply:
does "instructions" sound better?